src/search/SearchMessagesScreen.js

```diff
@@ -130,6 +154,50 @@ class SearchMessagesScreenInner extends PureComponent<Props, State> {
     this.handleQuerySubmit(e);
   };
 
+  fetchOlder = async () => {
+    if (
+      this.props.loading
+      || this.state.foundOldest
+      || this.state.isFetching
+      || this.isFetchingOlder
+    ) {
+      return;
+    }
+
+    invariant(this.state.messages !== null, 'must have atleast 1 message already fetched.');
+    this.isFetchingOlder = true;
+    try {
+      const { query } = this.state;
+      const anchor = this.state.messages[0].id;
+      const { messages: fetchedMessages, foundOldest } = await await this.props.dispatch(
+        fetchMessages({
+          anchor,
+          narrow: SEARCH_NARROW(this.state.query),
+          numBefore: config.messagesPerRequest,
+          numAfter: 0,
+        }),
+      );
+      this.setState(prevState => {
+        if (
+          prevState.query === query
+          && prevState.messages
+          && anchor === prevState.messages[0].id
+        ) {
+          // remove the anchor element, that gets fetched by default
+          fetchedMessages.pop();
+          return { messages: [...fetchedMessages, ...prevState.messages], foundOldest };
+        }
+        return {};
+      });
+    } finally {
+      this.isFetchingOlder = false;
+    }
+  };
+
+  fetchNewer = async () => {
+    // Implementing this should not be necessary.
```

Comment:
Let's have this throw — it should be impossible for it to get called, and that will let us know if that expected invariant fails.

Reply:
I don't think it is impossible for it to get called, in fact it does get called a lot of times, based on the condition for it to get triggered in `handleOutboundEvents.js`
```javascript
  if (innerHeight + scrollY >= offsetHeight - config.messageListThreshold) {
    props.fetchNewer();
  }
```
As far as I could understand, This gets called whenever we are close enough to the bottom (latest) of the messages in our search screen, so throwing an error would not be practical unless we change our conditions.

I have left this unchanged, let me know what you think about this.

Reply:
Mmm, I see, right. It's this callback's responsibility to see that there's nothing to do, just like how in `fetchOlder` we return earlier if `this.state.foundOldest`.

The comment should just make that clear, then. When I see "should not be necessary", I immediately wonder *why* it shouldn't be necessary — what assumption this code is relying on when it chooses not to do anything.

I'll revise that bit, and then go ahead and merge.